File: 728-self-dividing-numbers/self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
     ArrayList<Integer> r = new ArrayList<>();
     for(int i = left ; i <= right ; i++){
     int num = i;
     while(num!=0){
        int d = num%10;
        if(d == 0 || i%d != 0){
            break;
        }
        num = num/10;
     }
     if(num == 0){
        r.add(i);
        
     }
     }
     return r;
    }
}

File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> list[int]:
        ans = []
        for num in range(left , right+1):
            n = num
            valid = True
            while n > 0:
                d = n%10
                if d == 0 or num%d != 0:
                    valid = False
                    break
                n//=10
            if valid:
                ans.append(num)
        return ans

